File: Examples/PackageDiagram/Java/Banking/InMemoryAccountRepository.java
import java.util.HashMap;
import java.util.Map;

/** An in-memory `AccountRepository`. Part of the `Banking` module; depends on `Core`. */
final class InMemoryAccountRepository implements AccountRepository {
    private final Map<String, Account> storage = new HashMap<>();

    public Account find(String id) {
        return storage.get(id);
    }

    public void save(Account account) {
        storage.put(account.id, account);
    }
}

File: Examples/PackageDiagramDiff/Python/Before/Banking/in_memory_account_repository.py
from account import Account
from account_repository import AccountRepository


class InMemoryAccountRepository(AccountRepository):
    """An in-memory ``AccountRepository``. Part of the ``Banking`` module; depends on ``Core``."""

    def __init__(self):
        self._storage: dict[str, Account] = {}

    def find(self, id: str) -> Account | None:
        return self._storage.get(id)

    def save(self, account: Account) -> None:
        self._storage[account.id] = account
